tests/sentry/tasks/test_on_demand_metrics.py

```diff
@@ -362,14 +365,23 @@ def test_process_widget_specs(
             for mock_call in _set_cardinality_cache.mock_calls
         )
 
+    expected_state = ""
+    if not feature_flags:
+        expected_state = OnDemandExtractionState.DISABLED_PREROLLOUT
+    else:
+        expected_state = (
+            OnDemandExtractionState.ENABLED_ENROLLED
+            if expected_low_cardinality
+            else OnDemandExtractionState.DISABLED_HIGH_CARDINALITY
+        )
```

Comment:
This block comes from this one:
<img width="399" alt="image" src="https://github.com/getsentry/sentry/assets/44410/1c79edac-be0b-4757-97af-a428c2363e70">